capstone/src/test/java/com/google/sps/servlets/PostTest.java

```diff
@@ -0,0 +1,160 @@
+  private static final String POST_TEXT = "a great post";
+  private static final String CHALLENGE_NAME = "run 4 miles";
+  private static final String IMG = "";
+  private static final long TIMESTAMP = 123123123;
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig()
+              .setDefaultHighRepJobPolicyUnappliedJobPercentage(0));
+
+  private Post post;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    post = 
+      new Post(
+        4324344, /* postId */ 
+        AUTHOR_ID, /* authorId */ 
+        POST_TEXT, /* postText */
+        new ArrayList<Comment>(), /* comments */
+        CHALLENGE_NAME, /* challengeName */
+        TIMESTAMP, /* timestamp */
+        IMG, /* img */
+        new HashSet<String>() /* likes */);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+    post = null;
+  }
+
+  @Test
+  public void getPostTextTest() {
+    assertEquals(post.getPostText(), POST_TEXT);
+  }
+
+  @Test 
+  public void getImgtTest() {
+    assertEquals(post.getImg(), IMG);
+  }
+
+  @Test
+  public void getChallengeNameTest() {
+    assertEquals(post.getChallengeName(), CHALLENGE_NAME);
+  }
+
+  @Test 
+  public void getCommentsTest() {
+    assert post.getComments().size() == 0;
+  }
+
+  @Test
+  public void getLikesTest() {
+    assert post.getLikes().size() == 0;
+    post.addLike("user 1");
+    post.addLike("user 2");
+    HashSet<String> testLikes = new HashSet<String>();
+    testLikes.add("user 1");
+    testLikes.add("user 2");
+    
+    assert testLikes.containsAll(post.getLikes());
+  }
+
+  @Test
+  public void addCommentTest() {
+    Comment testComment1 =  
+      new Comment(
+        4324344, /* userId */ 
+        "a great comment", /* commentText */ 
+        "123123123" /* userId */ );
+    Comment testComment2 = 
+      new Comment(
+        55555555, /* userId */ 
+        "another great comment", /* commentText */ 
+        "09090909" /* userId */ );
+    post.addComment(testComment1);
+    post.addComment(testComment2);
+
+    assert post.getComments().size() == 2;
+    assertTrue(post.getComments().get(0).equals(testComment1));
+    assertTrue(post.getComments().get(1).equals(testComment2));
+  }
+
+  @Test
+  public void addLikeTest() {
+    post.addLike("user 1");
+    post.addLike("user 2");
+    HashSet<String> likes = post.getLikes();
+
+    assert likes.size() == 2;
+    assert likes.contains("user 1");
+    assert likes.contains("user 2");
+  }
+
+  @Test
+  public void removeLikeTest() {
+    post.addLike("test user 1");
+    post.addLike("test user 2");
+    HashSet<String> likes = post.getLikes();
+    assert likes.size() == 2;
+
+    post.removeLike("test user 1");
+    likes = post.getLikes();
+
+    assert likes.size() == 1;
+    assert likes.contains("test user 2");
+
+    post.removeLike("test user 2");
+    likes = post.getLikes();
+
+    assert likes.size() == 0;
+  }
```

Comment:
This should be 2 3 block tests:

`testRemoveLike` and `testRemoveLikeMultiple`

where you call removeLike once, or twice, and then check the value.

Having multiple places where you're asserting is less clean and obvious what's going wrong when you debug a failing test.